frontend/src/metabase/lib/colors/palette.ts

```diff
@@ -80,17 +80,27 @@ const aliases: Record<string, (palette: ColorPalette) => string> = {
   "accent7-dark": palette => shade(color(`accent7`, palette)),
 };
 
-export const color = (color: string, palette = colors) => {
+export function color(
+  colorName: keyof ColorPalette,
+  palette?: ColorPalette,
+): string;
+export function color(color: string, palette?: ColorPalette): string;
+export function color(color: any, palette: ColorPalette = colors) {
+  const fullPalette = {
+    ...colors,
+    ...palette,
+  };
```

Comment:
This allows `palette` to contain partial colors. Making places that rely on custom using this function easier, otherwise, we need to construct the whole color palette which is cumbersome, and why we need to do that in the first place since the main colors are define in this file.

I think it's this file/function's responsibility to merge `palette` with existing `colors`